File: clients/metorial-js/packages/core/src/mt_2025_01_01_pulsar/resources/management/instance/files/update.ts
import { mtMap } from '@metorial/util-resource-mapper';

export type ManagementInstanceFilesUpdateOutput = {
  id: string;
  status: 'active' | 'deleted';
  fileName: string;
  fileSize: number;
  fileType: string;
  title: string | null;
  purpose: { name: string; identifier: string };
  createdAt: Date;
  updatedAt: Date;
};

export let mapManagementInstanceFilesUpdateOutput =
  mtMap.object<ManagementInstanceFilesUpdateOutput>({
    id: mtMap.objectField('id', mtMap.passthrough()),
    status: mtMap.objectField('status', mtMap.passthrough()),
    fileName: mtMap.objectField('file_name', mtMap.passthrough()),
    fileSize: mtMap.objectField('file_size', mtMap.passthrough()),
    fileType: mtMap.objectField('file_type', mtMap.passthrough()),
    title: mtMap.objectField('title', mtMap.passthrough()),
    purpose: mtMap.objectField(
      'purpose',
      mtMap.object({
        name: mtMap.objectField('name', mtMap.passthrough()),
        identifier: mtMap.objectField('identifier', mtMap.passthrough())
      })
    ),
    createdAt: mtMap.objectField('created_at', mtMap.date()),
    updatedAt: mtMap.objectField('updated_at', mtMap.date())
  });

export type ManagementInstanceFilesUpdateBody = { title?: string | undefined };

export let mapManagementInstanceFilesUpdateBody =
  mtMap.object<ManagementInstanceFilesUpdateBody>({
    title: mtMap.objectField('title', mtMap.passthrough())
  });

File: clients/metorial-py/packages/core/mt_2025_01_01_pulsar/resources/dashboard/instance/files/update.py

from dataclasses import dataclass
from typing import Any, Dict, List, Optional, Union
from datetime import datetime

@dataclass
class DashboardInstanceFilesUpdateOutput:
    object: str
    id: str
    status: str
    file_name: str
    file_size: float
    file_type: str
    purpose: Dict[str, Any]
    created_at: datetime
    updated_at: datetime
    title: Optional[str] = None


from typing import Any, Dict
from datetime import datetime

def mapDashboardInstanceFilesUpdateOutput(data: Dict[str, Any]) -> DashboardInstanceFilesUpdateOutput:
    return DashboardInstanceFilesUpdateOutput(
    object=data.get('object'),
    id=data.get('id'),
    status=data.get('status'),
    file_name=data.get('file_name'),
    file_size=data.get('file_size'),
    file_type=data.get('file_type'),
    title=data.get('title'),
    purpose={
        "name": data.get('purpose', {}).get('name'),
        "identifier": data.get('purpose', {}).get('identifier')
    },
    created_at=data.get('created_at') and datetime.fromisoformat(data.get('created_at')),
    updated_at=data.get('updated_at') and datetime.fromisoformat(data.get('updated_at'))
    )


from dataclasses import dataclass
from typing import Any, Dict, List, Optional, Union
from datetime import datetime

@dataclass
class DashboardInstanceFilesUpdateBody:
    title: Optional[str] = None


from typing import Any, Dict
from datetime import datetime

def mapDashboardInstanceFilesUpdateBody(data: Dict[str, Any]) -> DashboardInstanceFilesUpdateBody:
    return DashboardInstanceFilesUpdateBody(
    title=data.get('title')
    )

